### PR TITLE
fix/AvatarFavicon to TS

### LINKED ISSUE #19117: Migrate components to TS: AvatarFavicon
### Description
Migrate `AvatarFavicon` component to TypeScript

### Technical Details

- Convert Objects in `SNAKE_CASE` to enums in `PascalCase` (name and keys)
- Ensure enums use singular not plural e.g. `Size` not `Sizes`
- Update file name from `component-name.contants.js` to `component-name.types.ts` and add types
- Update file name from `component-name.js` to `component-name.tsx` and add types
- Update file name from `component-name.test.js` to `component-name.test.tsx`
- Update file name from `component-name.stories.js` to `component-name.stories.tsx`
- Update all instances of component objects to enums
- Update all documentation

### Acceptance Criteria

- Component has been converted to TypeScript
- Stories, documentation and tests have been updated inside of the `component-name` folder
- Before/After screen cast of component stories in storybook to show no visual regression after migration

### Difficulty: Intermediate

Good first issue for: External contributors who are familiar with TypeScript and want to contribute to improving the stability of the extension

### PR DESCRIPTION
**Pull request by dhruvv173**
## Explanation

- Fixes #19117 

### Before

https://github.com/MetaMask/metamask-extension/assets/79097544/3b7f06cb-05fd-4f40-9b6e-470289008f96

### After

https://github.com/MetaMask/metamask-extension/assets/79097544/4d2582f0-2f90-4381-8644-71d0781357dd

image

## Manual Testing Steps

- Check storybook controls for `AvatarFavicon`
- `yarn jest ui/components/component-library/avatar-favicon`

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/component-library/avatar-favicon/README.mdx [hunk 1]
```diff
@@ -14,44 +14,36 @@ The `AvatarFavicon` is an image component that renders an icon that is provided
...
-component props
-
-<ArgsTable of={AvatarBase} />
-
 ### Size
 
-Use the `size` prop to set the size of the `AvatarFavicon`.
-
-Optional: `AVATAR_FAVICON_SIZES` from `./ui/components/component-library` object can be used instead of `Size`
+Use the `Size` prop or the `AvatarFaviconSize` enum from `../../component-library` to change the size of `AvatarFavicon`
```
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-favicon/README.mdx:23]: Update sentence to refer to the `size` prop and enum
```suggestion
Use the `size` prop and the `AvatarFaviconSize` enum from `../../component-library` to change the size of `AvatarFavicon`
```

File: ui/components/component-library/avatar-favicon/avatar-favicon.types.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+   */
+  name: string;
+  /**
+   * Props for the fallback icon. All Icon props can be used
+   */
+  fallbackIconProps?: IconProps<'span'>;
+  /**
+   * The size of the AvatarFavicon
+   * Possible values could be 'Size.XS' 16px, 'Size.SM' 24px, 'Size.MD' 32px, 'Size.LG' 40px, 'Size.XL' 48px
+   * Defaults to Size.MD
```
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-favicon/avatar-favicon.types.ts:31]:
```suggestion
   * Possible values could be 'AvatarFaviconSize.XS' 16px, 'AvatarFaviconSize.SM' 24px, 'AvatarFaviconSize.MD' 32px, 'AvatarFaviconSize.LG' 40px, 'AvatarFaviconSize.XL' 48px
   * Defaults to AvatarFaviconSize.MD
```

**Review by georgewrmarshall — CHANGES_REQUESTED** [review]
Looking good! Left some small suggestions I still need to do a full review but will leave these comments and review again tomorrow

File: ui/components/component-library/avatar-favicon/README.mdx [hunk 3]
```diff
@@ -14,44 +14,36 @@ The `AvatarFavicon` is an image component that renders an icon that is provided
 
 ## Props
 
-The `AvatarFavicon` accepts all props below as well as all [Box](/docs/components-ui-box--default-story#props) component props
+The `AvatarFavicon` accepts all props below as well as all [Box](/docs/components-componentlibrary-box--docs#props) component props
```
[the remark this answers is not in the document]
Comment (dhruvv173) [reply ui/components/component-library/avatar-favicon/README.mdx]: removed [`here`](https://github.com/MetaMask/metamask-extension/pull/20430/commits/6774ca92db19fd39d3c0a46652d3a410c89eff0d)

File: ui/components/component-library/avatar-favicon/README.mdx (hunk 1 above)
Comment (dhruvv173) [reply ui/components/component-library/avatar-favicon/README.mdx]: updated [`here`](https://github.com/MetaMask/metamask-extension/pull/20430/commits/6774ca92db19fd39d3c0a46652d3a410c89eff0d)

File: ui/components/component-library/avatar-favicon/avatar-favicon.types.ts (hunk 2 above)
Comment (dhruvv173) [reply ui/components/component-library/avatar-favicon/avatar-favicon.types.ts]: updated [`here`](https://github.com/MetaMask/metamask-extension/pull/20430/commits/6774ca92db19fd39d3c0a46652d3a410c89eff0d)

**Review by garrettbear — DISMISSED** [review]
LGTM now! Just waiting on all tests to pass.

**Review by garrettbear — DISMISSED** [review]

**Comment by dhruvv173** [comment]
> Ahh found the issue:
> 
> ### Before
> ```
> import { BorderColor } from 'ui/helpers/constants/design-system';
> import { AvatarBaseStyleUtilityProps } from '../avatar-base/avatar-base.types';
> ```
> 
> ### After
> ```
> import { BorderColor } from '../../../helpers/constants/design-system';
> import type { AvatarBaseStyleUtilityProps } from '../avatar-base/avatar-base.types';
> ```

my bad I missed it, thank you for fixing it!

File: ui/components/component-library/avatar-account/README.mdx [hunk 4]
```diff
@@ -69,7 +69,7 @@ Use the required `address` for generating images
 </Canvas>
 
 ```jsx
-import { AvatarAccount, AvatarAccountVariant } from '../ui/component-library';
+import { AvatarAccount, AvatarAccountVariant } from '../../component-library';
 
 <AvatarAccount variant={AvatarAccountVariant.Jazzicon} address="0x5CfE73b6021E818B776b421B1c4Db2474086a7e1" />
 <AvatarAccount variant={AvatarAccountVariant.Blockies} address="0x0" />
```
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-account/README.mdx:75]: Updating some incorrect component paths

File: ui/components/component-library/avatar-icon/README.mdx [hunk 5]
```diff
@@ -83,7 +83,7 @@ Use the `color` and `backgroundColor` props with the `IconColor` and `Background
 
 ```jsx
 import { BackgroundColor, IconColor } from '../../../helpers/constants/design-system';
-import { AvatarIcon } from '../ui/component-library';
+import { AvatarIcon } from '../../component-library';
```
Comment (georgewrmarshall) [inline ui/components/component-library/avatar-icon/README.mdx:87]: Updating some incorrection component paths in docs

File: ui/components/component-library/button/README.mdx [hunk 6]
```diff
@@ -67,7 +67,7 @@ Possible sizes include:
 
 ```jsx
 import { Size } from '../../../helpers/constants/design-system';
-import { Button } from '../ui/component-library';
+import { Button } from '../../component-library';
```
Comment (georgewrmarshall) [inline ui/components/component-library/button/README.mdx:70]: Updating some incorrect component paths in docs

**Review by georgewrmarshall — APPROVED** [review]
LGTM! I made [some small updates to the docs](https://github.com/MetaMask/metamask-extension/pull/20430/commits/877846353046684804eabfe42090bfc6a7a7d850) and thought it would be low hanging fruit and wouldn't increase the PR to much if I caught the other instances of the incorrect component path in other component docs. Great work 💯 Thanks for your contribution @dhruvv173 

Pulled branch and checked
- storybook docs ✅ 
- storybook stories and controls ✅ 
- console for warnings or errors ✅ 
- passes storybook accessibility tests ✅ 
- Searched `AVATAR_FAVICON_SIZES` to ensure no instances ✅ 

No more incorrect component paths in docs

Screenshot 2023-08-15 at 2 01 25 PM

**Review by garrettbear — APPROVED** [review]

**Pull request merged**